Solve in parallel

anne-hids opened the pull request:
This update to the SolverModule makes it possible to solve in parallel. The code can be tested using the branch "SolveParallelExportWorld" in SBooScripts.

jormercury committed:
SBoo update to 2024.11.0 (#177)

Since the initial release for verification with SimpleBox 4 in excel a
lot more development has happened, specifically related to the solvers.
These updates are intended to part of release 2024.11.0

anne-hids committed:
Adjusted solver to work for parallel solving

anne-hids committed:
Update R6SolverModule.R

anne-hids committed:
Update R6SolverModule.R

anne-hids committed:
Removed browser statements

anne-hids committed:
Changed filetype of World and LHS samples from RData to RDS

jaap-slootweg reviewed (approved):
Nicely done, might lead to additional niceties when further processing is needed, like for graphics etc. We'll see.

anne-hids committed:
Bugfix

anne-hids committed:
Update R6SolverModule.R

anne-hids committed:
Bugfix

anne-hids committed:
Solver module update (#196)

Changes in this pull request:

- No more s_ solver functions: there is only one file with solver
functions, names fGeneral.R
- In the solvermodule, the solver functions from fGeneral are used
depending on the variables given by the user
- Solving steady state deterministically, steady state
probabilistically, dynamic deterministically and dynamic
probabilistically happens inside the solver module
- The concentration module was deleted, instead there is a function that
is called from the solver module which calculates concentrations
- Standard plot functions were added to SBcore, reachable via:
World$PlotConcentration() and World$PlotSolution(). These functions
could possibly use some more refinement. The actual functions used by
the core for plotting are in fPlotFunctions.R.
- Testscripts were added to test the new functionality of SBoo
- The existing solver vignettes were deleted. Instead, there is one
solver vignette: x Solver use.rmd

anne-hids committed:
Update CITATION.cff

jormercury committed:
test build Rd files (does not pass yet)

jormercury committed:
Update CITATION.cff

jormercury committed:
Update CITATION.cff

jormercury committed:
Update CITATION.cff

jormercury committed:
Update CITATION.cff

jormercury committed:
Update metadata (#211)

Update to the citation file. Now includes authors, license, keywords,
abstract and version.

jormercury committed:
Merge branch 'development' into Documentation_build

jormercury committed:
Update CITATION.cff

jormercury committed:
Update of documentation files for latest package test build (#212)

Update of documentation files for latest package test build

jormercury committed:
Merge branch 'main' into development

anne-hids committed:
Merge branch 'development' into SolveParallelExportWorld
